pkg/controller/servicemesh/webhook/controller.go

```diff
@@ -0,0 +1,247 @@
+		logger.Info("reconciling MutatingWebhookConfiguration")
+		// get current webhook config
+		currentConfig := &v1beta1.MutatingWebhookConfiguration{}
+		err := r.Client.Get(context.TODO(), request.NamespacedName, currentConfig)
+		if err != nil {
+			log.Info("MutatingWebhookConfiguration does not exist yet. No action taken")
+			return reconcile.Result{}, nil
+		}
+		namespace := request.Name[len(sidecarInjectorWebhookConfigNamePrefix):]
+		// retrieve root certificate from secret
+		caRoot, err := r.getCABundle(namespace, sidecarInjectorServiceAccountSecretName)
+		if err != nil {
+			logger.Info("could not get secret: " + err.Error())
+			return reconcile.Result{}, nil
+		}
+		// update caBundle if it doesn't match what's in the secret
+		newConfig := currentConfig.DeepCopyObject().(*v1beta1.MutatingWebhookConfiguration)
+		for i := range newConfig.Webhooks {
+			updated = injectCABundle(&newConfig.Webhooks[i].ClientConfig, caRoot) || updated
+		}
+		config = newConfig
+	} else if strings.HasPrefix(request.Name, galleyWebhookConfigNamePrefix) {
+		logger.Info("reconciling ValidatingWebhookConfiguration")
+		// get current webhook config
+		currentConfig := &v1beta1.ValidatingWebhookConfiguration{}
+		err := r.Client.Get(context.TODO(), request.NamespacedName, currentConfig)
+		if err != nil {
+			log.Info("ValidatingWebhookConfiguration does not exist yet. No action taken")
+			return reconcile.Result{}, nil
+		}
+		namespace := request.Name[len(galleyWebhookConfigNamePrefix):]
+		// retrieve root certificate from secret
+		caRoot, err := r.getCABundle(namespace, galleyServiceAccountSecretName)
+		if err != nil {
+			logger.Info("could not get secret: " + err.Error())
+			return reconcile.Result{}, nil
+		}
+		// update caBundle if it doesn't match what's in the secret
+		newConfig := currentConfig.DeepCopyObject().(*v1beta1.ValidatingWebhookConfiguration)
+		for i := range newConfig.Webhooks {
+			updated = injectCABundle(&newConfig.Webhooks[i].ClientConfig, caRoot) || updated
+		}
+		config = newConfig
+	} else {
+		// This shouldn't happen.
+		return reconcile.Result{}, errors.New("tried to reconcile unknown webhook configuration")
+	}
+
+	if updated {
+		err := r.Client.Update(context.TODO(), config)
+		if err != nil {
+			return reconcile.Result{}, errors2.Wrap(err, "failed to update CABundle")
+		}
+		logger.Info("CABundle updated")
+		return reconcile.Result{}, nil
+	}
+
+	logger.Info("Correct CABundle already present. Ignoring")
+	return reconcile.Result{}, nil
+}
+
+func (r *reconciler) getCABundle(namespace string, name string) ([]byte, error) {
+	// get secret
+	secret := &corev1.Secret{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: namespace,
+		Name:      name,
+	}, secret)
+	if err != nil {
+		return nil, err
+	}
+	caBundle, ok := secret.Data[rootCertKey]
+	if !ok {
+		return nil, errors.New("secret " + namespace + "/" + name + " does not contain root certificate under key " + rootCertKey)
```

Comment:
Use string formatting instead of concatenation?

Reply:
Will do